download-maven-plugin/src/main/java/com/googlecode/download/maven/plugin/internal/cache/Index.java

```diff
@@ -0,0 +1,32 @@
+package com.googlecode.download.maven.plugin.internal.cache;
+
+/**
+ * Cache index.
+ * @author Paul Polishchuk
+ * @since 1.3.1
+ */
+interface Index {
+
+    /**
+     * Adds given path to the index using url parameter as a key.
+     * @param url index key
+     * @param path index value
+     */
+    void put(String url, String path);
```

Comment:
Strong typing is better code in general. If we know we're storing the location of a File, it seems better to pass the File. Same thing for the URL, it's better to type it as URL (or URI since URL has some performance bottlenecks).